src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_ILocalFunctionStatement.cs

```diff
@@ -458,5 +471,85 @@ void M(int p)
 
             VerifyOperationTreeAndDiagnosticsForTest<LocalFunctionStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
         }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void TestInvalidLocalFunction_ExpressionAndBlockBody()
+        {
+            string source = @"
+class C
+{
+    void M(int p)
+    {
+        /*<bind>*/object F() => new object(); { return new object(); }/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+ILocalFunctionOperation (Symbol: System.Object F()) (OperationKind.LocalFunction, Type: null) (Syntax: 'object F()  ... w object();')
+  Expression Body: 
+    IBlockOperation (1 statements) (OperationKind.Block, Type: null) (Syntax: '=> new object()')
+      IReturnOperation (OperationKind.Return, Type: null, IsImplicit) (Syntax: 'new object()')
+        ReturnedValue: 
+          IObjectCreationOperation (Constructor: System.Object..ctor()) (OperationKind.ObjectCreation, Type: System.Object) (Syntax: 'new object()')
+            Arguments(0)
+            Initializer: 
+              null
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // file.cs(6,49): error CS0127: Since 'C.M(int)' returns void, a return keyword must not be followed by an object expression
+                //         /*<bind>*/object F() => new object(); { return new object(); }/*</bind>*/;
+                Diagnostic(ErrorCode.ERR_RetNoObjectRequired, "return").WithArguments("C.M(int)").WithLocation(6, 49),
+                // file.cs(6,26): warning CS8321: The local function 'F' is declared but never used
+                //         /*<bind>*/object F() => new object(); { return new object(); }/*</bind>*/;
+                Diagnostic(ErrorCode.WRN_UnreferencedLocalFunction, "F").WithArguments("F").WithLocation(6, 26)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<LocalFunctionStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void TestInvalidLocalFunction_BlockAndExpressionBody()
```

Comment:
>TestInvalidLocalFunction_BlockAndExpressionBody [](start = 20, length = 47)

It doesn't look like we test SemanticModel APIs for this scenario. Please make sure we have existing tests covering the case of both bodies present.